core-bundle/src/ServiceAnnotation/Hook.php

```diff
@@ -29,15 +29,8 @@
  */
 final class Hook implements ServiceTagInterface
 {
-    /**
-     * @var string
-     */
-    public $value;
-
-    /**
-     * @var int
-     */
-    public $priority;
+    public string $value;
+    public int $priority = 0;
```

Comment:
Is this correct or does the default value have to be `null`?

Reply:
The doc used to be `int|null`, but that lead to an `AnnotationException`: https://github.com/contao/contao/pull/764
But it is supposed to be `null` if not defined otherwise. So I think `public int $priority;` is correct.

But I think you will need to test if hook registrations with

```php
/**
  * @Hook(…)
  */
```

and

```php
/**
  * @Hook(…, priority=-10)
  */
```

after the change.

Reply:
> So I think `public int $priority;` is correct.

No, it is not. It will lead to an `Error: Typed property Contao\CoreBundle\ServiceAnnotation\Hook::$priority must not be accessed before initialization` error if there is no default value, therefore the question is does the default value have to be `null` or can it also be `0`? @contao/developers

Reply:
Well, if you say it must be an `int`, then it must be an `int` :D It cannot be `null`. That would only be valid if you specified `?int`.

Reply:
Yes, that's what I meant to write actually 🙈. I meant it should be tested with `public ?int $priority;` since the phpdoc type definition `int|null` lead to an error with the annotation parser in the past.

Reply:
> Well, if you say it must be an `int`, then it must be an `int` :D It cannot be `null`.

If we need the default to be `null`, we would have to change the type to `public ?int $priority = null`. But I think `null` is equivalent to `0` anyway, because any service without a priority will implicitly have `o`, won't it?

Reply:
As discussed with @aschempp, it has to be `public ?int $priority = null` (see #3444).